prusti-common/src/vir/ast/function.rs

```diff
@@ -62,25 +62,36 @@ impl Function {
             }
         })
     }
+
+    pub fn apply(&self, args: Vec<Expr>) -> Expr {
+        Expr::FuncApp(
+            self.name.to_string(),
+            args,
+            self.formal_args.clone(),
+            self.return_type.clone(),
+            Position::default(),
+        )
+    }
 }
 
 pub fn compute_identifier(name: &str, formal_args: &[LocalVar], return_type: &Type) -> String {
     let mut identifier = name.to_string();
     // Include the signature of the function in the function name
     identifier.push_str("__$TY$__");
-    fn type_name(typ: &Type) -> &str {
+    fn type_name(typ: &Type) -> String {
         match typ {
-            Type::Int => "$int$",
-            Type::Bool => "$bool$",
-            Type::TypedRef(ref name) => name,
-            Type::Domain(ref name) => name,
+            Type::Int => "$int$".to_string(),
+            Type::Bool => "$bool$".to_string(),
+            Type::TypedRef(ref name) => name.to_string(),
+            Type::Domain(ref name) => name.to_string(),
+            Type::Snapshot(ref name) => format!("Snap${}", name),
```

Comment:
I am wondering whether it would not be better to store the snapshot type name as `Snap$<name>` so that we can return a reference here. In case we ever need to extract the `name`, we could subslice, which is a much cheaper operation than creating a new string.